xds/internal/balancer/ringhash/config.go

```diff
@@ -0,0 +1,48 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package ringhash
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"google.golang.org/grpc/serviceconfig"
+)
+
+// Name is the name of the ring_hash balancer.
+const Name = "ring_hash_experimental"
```

Comment:
I think using this const is the right way. And each balancer should have an exported `Name`.

One convenience we get is that, if we decide to remove the `_experimental` suffix later, we only need to change it at one place.

Reply:
I agree about the convenience of having to change at a single place. But the dependencies added between balancer implementations worries me. How about a `balancernames` package which contains only balancer names? So, we will have `balancernames.RoundRobin` and `balancerNames.ClusterResolver` etc.

Reply:
Why is the dependencies between balancer packages a problem?
If a balancer uses another one as children, it may need to import that (either for the name, or for the config). This sounds normal to me.

Reply:
I feel they are a problem because they can be avoided. If one policy needs another one as a child, all it needs to know is the policy name and the configuration format. Given how complicated our LB policy trees are becoming, I feel that it would be good to avoid dependencies which *can* be avoided.

Reply:
> all it needs to know is the policy name and the configuration format

I think the better way to get the policy is to import the package, and use the exported variable `Name`, instead of making another const and hardcode the name.

So I actually think the dependencies between is by design. And not all policies import the children.
- `cluster_resolver` needs to, and should, import `weightedtarget` and `ringhash`. Because the tree is predefined, and there's no case where `cluster_resolver` is used, but not `weightedtarget` and `ringhash`.
- `weightedtarget` doesn't import any child policy, because it's a general purpose policy, and its children can be anything. The child policy names come from the resolver update.